app/controllers/admin/users_controller.rb

```diff
@@ -1,6 +1,7 @@
 class Admin::UsersController < ApplicationController
   before_action :authorize_admin!
   before_action :authorize_super_admin!, only: [ :update, :destroy ]
+  after_action :remove_user_locks, only: [:destroy]
```

Comment:
In that case, agreed a better fix is to add that association to the User object, and have it handle the cascading soft delete.